crates/doctor/src/rustlang/target.rs

```diff
@@ -0,0 +1,177 @@
+use anyhow::Result;
+use async_trait::async_trait;
+
+use crate::{Diagnosis, Diagnostic, PatientApp, Treatment};
+
+/// VersionDiagnostic detects problems with the app manifest version field.
+#[derive(Default)]
+pub struct TargetDiagnostic;
+
+#[async_trait]
+impl Diagnostic for TargetDiagnostic {
+    type Diagnosis = TargetDiagnosis;
+
+    async fn diagnose(&self, patient: &PatientApp) -> Result<Vec<Self::Diagnosis>> {
+        // TODO: this, down to the "does the app use Rust" check, probably ought to move up to the Rust level
+        // but we can defer this until we have more Rust diagnoses
+        let path = &patient.manifest_path;
+        let manifest = spin_loader::local::raw_manifest_from_file(&path)
+            .await?
+            .into_v1();
+        let uses_rust = manifest.components.iter().any(|c| {
+            c.build
+                .as_ref()
+                .map(|b| b.command.starts_with("cargo"))
+                .unwrap_or_default()
+        });
+
+        if uses_rust {
+            // does any component contain a build command with `cargo` as the program?
+            // if so, run rustup target list --installed and:
+            // - if rustup is not present, return RustNotInstalled
+            // - if rustup is present but the list does not contain wasm32-wasi, return WasmTargetNotInstalled
+            // - if the list does contain wasm32-wasi, return an empty list
+            // NOTE: this does not currently check against the Rust SDK MSRV - that could
+            // be a future enhancement or separate diagnosis, but at least the Rust compiler
+            // should give a clear error for that!
+            let target_list_output = tokio::process::Command::new("rustup")
+                .args(["target", "list", "--installed"])
+                .output()
+                .await;
+            let diagnosis = match target_list_output {
+                Err(e) => {
+                    if e.kind() == std::io::ErrorKind::NotFound {
+                        vec![TargetDiagnosis::RustNotInstalled]
+                    } else {
+                        anyhow::bail!(e)
```

Comment:
Can we add context of the command we're running? The error message may only make sense if the user knows we're running `rustup target list --installed`.